docker/run.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ "$2" == "" ]; then
+    echo "Usage: run.sh --map-path path_to_map_files"
+    exit 1
+fi
+
+# Get the map path
+while [ "$1" != "" ]; do
+    case "$1" in
+    --map-path)
+        MAP_PATH="$2"
+        shift
+        ;;
+    *)
+        args+=("$1")
+        ;;
+    esac
+    shift
+done
+
+# Launch Autoware with custom map
+echo "Launching Autoware with the map >> "${MAP_PATH}
+docker run -it --rm --net=host --gpus all -e DISPLAY=$DISPLAY -v /tmp/.X11-unix/:/tmp/.X11-unix \
```

Comment:
With rocker a very nice feature was the ability to setup the container to have the same user id and group as on the host, which allowed to share directories (autoware code, maps, neural networks) between host and container. Is it planned to have a replacement for that functionality with docker compose or something else?

Reply:
Although rocker has the good feature of user login it introduces extra tooling for running the containers.
Same functionality can be achieved with the new **dev_entrypoint.sh**  and **run.sh** script.

dev_entrypoint.sh
```shell
#!/bin/bash

# Get the user ID and group ID of the local user
USER_ID=${LOCAL_UID}
USER_NAME=${LOCAL_USER}
GROUP_ID=${LOCAL_GID}
GROUP_NAME=${LOCAL_GROUP}
echo "Starting with user: $USER_NAME >> UID $USER_ID, GID: $GROUP_ID"

# Create group and user with GID/UID
groupadd -g $GROUP_ID  $GROUP_NAME
useradd -u $USER_ID -g $GROUP_ID -s /bin/bash -m -d /home/$USER_NAME $USER_NAME

# Add sudo privileges to the user
echo "$USER_NAME ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers

# Source ROS2
source "/opt/ros/$ROS_DISTRO/setup.bash"

# Execute the command as the user
exec /usr/sbin/gosu $USER_NAME "$@"
```

Reply:
Nice!

Otherwise yes, I agree with removing rocker.